<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BDDF Download</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Bosdyn API Client Common Arguments</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Hostname</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_hostname">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;robot ipv4 address&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="placeholderText">
           <string>192.168.80.3</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_ping">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>ping</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Username</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_username"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Password </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_password">
          <property name="echoMode">
           <enum>QLineEdit::PasswordEchoOnEdit</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QPushButton" name="btn_check_timesync">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Check timesync</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_timesync">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QPushButton" name="btn_check_data_buffer">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Check data buffer size</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_check_data_buffer">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;A timespan is {val} or {val}-{val} where:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;{val} has one of these formats:&lt;/p&gt;&lt;p&gt; - yyyymmdd_hhmmss  (e.g., 20200120_120000)&lt;/p&gt;&lt;p&gt; - yyyymmdd         (e.g., 20200120)&lt;/p&gt;&lt;p&gt; -  {n}d    {n} days ago&lt;/p&gt;&lt;p&gt; -  {n}h    {n} hours ago&lt;/p&gt;&lt;p&gt; -  {n}m    {n} minutes ago&lt;/p&gt;&lt;p&gt; -  {n}s    {n} seconds ago&lt;/p&gt;&lt;p&gt; - nnnnnnnnnn[.nn]       (e.g., 1581869515.256)  Seconds since epoch&lt;/p&gt;&lt;p&gt; - nnnnnnnnnnnnnnnnnnnn  Nanoseconds since epoch&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;For example:&lt;/p&gt;&lt;p&gt;  '5m'                    From 5 minutes ago until now.&lt;/p&gt;&lt;p&gt;  '20201107-20201108'     All of 2020/11/07.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Timespan (must be &lt;60 minutes)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="le_val">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;A timespan is {val} or {val}-{val} where:&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;{val} has one of these formats:&lt;/p&gt;&lt;p&gt; - yyyymmdd_hhmmss  (e.g., 20200120_120000)&lt;/p&gt;&lt;p&gt; - yyyymmdd         (e.g., 20200120)&lt;/p&gt;&lt;p&gt; -  {n}d    {n} days ago&lt;/p&gt;&lt;p&gt; -  {n}h    {n} hours ago&lt;/p&gt;&lt;p&gt; -  {n}m    {n} minutes ago&lt;/p&gt;&lt;p&gt; -  {n}s    {n} seconds ago&lt;/p&gt;&lt;p&gt; - nnnnnnnnnn[.nn]       (e.g., 1581869515.256)  Seconds since epoch&lt;/p&gt;&lt;p&gt; - nnnnnnnnnnnnnnnnnnnn  Nanoseconds since epoch&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;For example:&lt;/p&gt;&lt;p&gt;  '5m'                    From 5 minutes ago until now.&lt;/p&gt;&lt;p&gt;  '20201107-20201108'     All of 2020/11/07.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>5m</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cb_use_date_time">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string/>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>Use Date/Time Selector</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QDateTimeEdit" name="dte_1">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="dateTime">
           <datetime>
            <hour>0</hour>
            <minute>0</minute>
            <second>0</second>
            <year>2021</year>
            <month>1</month>
            <day>1</day>
           </datetime>
          </property>
          <property name="displayFormat">
           <string>yyyy/MM/dd hh:mm:ss</string>
          </property>
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>----&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QDateTimeEdit" name="dte_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="dateTime">
           <datetime>
            <hour>23</hour>
            <minute>59</minute>
            <second>0</second>
            <year>2021</year>
            <month>1</month>
            <day>1</day>
           </datetime>
          </property>
          <property name="time">
           <time>
            <hour>23</hour>
            <minute>59</minute>
            <second>0</second>
           </time>
          </property>
          <property name="displayFormat">
           <string>yyyy/MM/dd hh:mm:ss</string>
          </property>
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="cb_append_timespan">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Append timespan to filename</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Channel(s):</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_channel">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Separate each channel by a space.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="inputMask">
           <string/>
          </property>
          <property name="placeholderText">
           <string>channel.1 channel.2</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="2" rowspan="3">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Operator Comments</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pb_get_events_comments">
        <property name="text">
         <string>Get all comments</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="lw_comments">
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>0</height>
         </size>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="selectionRectVisible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="0" colspan="2">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>bddf_download.py arguments</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_select_all">
          <property name="text">
           <string>Select All</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="te_command">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; text-decoration: underline;&quot;&gt;Example:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;python3 bddf_download.py HOSTNAME -T 20210115_131800-20210115_140100 --username USER --password PASSWORD -o ~/Documents/download.bddf&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Waiting for input...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0" colspan="2">
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Output filepath  </string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_output_filepath">
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>~/Documents/bddf_download_gui</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_save_as">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Use the text inputs to the left or &amp;quot;Save As&amp;quot; to specify output filepath and filename before download.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>Save As</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Output filename</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="le_output_filename">
          <property name="text">
           <string>download.bddf</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_download">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start download.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(115, 210, 22);</string>
          </property>
          <property name="text">
           <string>Download</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QLabel" name="label_filesize">
          <property name="text">
           <string>Download progress:</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QProgressBar" name="pb">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <action name="action_debug">
   <property name="text">
    <string>debug</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WindowShortcut</enum>
   </property>
  </action>
  <action name="action_default">
   <property name="text">
    <string>DEFAULT</string>
   </property>
  </action>
  <action name="action_add_new_profile">
   <property name="text">
    <string>Add New Profile</string>
   </property>
  </action>
  <action name="action_info">
   <property name="text">
    <string>info</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="action_edit_config">
   <property name="text">
    <string>Edit config.ini</string>
   </property>
  </action>
  <action name="action_reload_config">
   <property name="text">
    <string>Reload config.ini</string>
   </property>
  </action>
  <action name="actionDEFAULT">
   <property name="text">
    <string>DEFAULT</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
